smarts/core/smarts.py

```diff
@@ -672,6 +672,13 @@ def matches_provider_action_spaces(agent_id, action_spaces):
             interface = self._agent_manager.agent_interface_for_agent_id(agent_id)
             return interface.action_space in action_spaces
 
+        # Check if actions are on active ego agents
+        for agent_id, action in ego_agent_actions.items():
```

Comment:
If only the keys are needed you can do:
```suggestion
        for agent_id in ego_agent_actions:
```
or 
```suggestion
        for agent_id in ego_agent_actions.keys():
```